subsys/fs/CMakeLists.txt

```diff
@@ -8,8 +9,10 @@ if(CONFIG_FILE_SYSTEM)
   zephyr_library_sources_ifdef(CONFIG_FILE_SYSTEM_NFFS   nffs_fs.c)
   zephyr_library_sources_ifdef(CONFIG_FILE_SYSTEM_SHELL  shell.c)
 
-  zephyr_library_link_libraries_ifdef(CONFIG_FAT_FILESYSTEM_ELM ELMFAT)
-  zephyr_library_link_libraries_ifdef(CONFIG_FILE_SYSTEM_NFFS   NFFS)
+  zephyr_library_link_libraries(FS)
+
+  target_link_libraries_ifdef(CONFIG_FAT_FILESYSTEM_ELM FS INTERFACE ELMFAT)
+  target_link_libraries_ifdef(CONFIG_FILE_SYSTEM_NFFS   FS INTERFACE NFFS)
```

Comment:
Shouldn't these two lines be in the CMakeList.txt of NFFS and ELMFAT respectively?

Reply:
Maybe, but they can't because FS can only be modified in the CMakeLists.txt directory that created it.